src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -0,0 +1,827 @@
+        /// of the same type has not already been registered.
+        /// </summary>
+        public static IGraphQLBuilder TryRegister<TService>(this IGraphQLBuilder graphQLBuilder, Func<IServiceProvider, TService> implementationFactory, ServiceLifetime serviceLifetime)
+            where TService : class
+            => graphQLBuilder.TryRegister(typeof(TService), implementationFactory ?? throw new ArgumentNullException(nameof(implementationFactory)), serviceLifetime);
+
+        /// <summary>
+        /// Registers <paramref name="implementationInstance"/> as type <typeparamref name="TService"/> with the dependency injection provider
+        /// if a service of the same type has not already been registered.
+        /// </summary>
+        public static IGraphQLBuilder TryRegister<TService>(this IGraphQLBuilder graphQLBuilder, TService implementationInstance)
+            where TService : class
+            => graphQLBuilder.TryRegister(typeof(TService), implementationInstance ?? throw new ArgumentNullException(nameof(implementationInstance)));
+
+        /// <inheritdoc cref="IGraphQLBuilder.Configure{TOptions}(Action{TOptions, IServiceProvider})"/>
+        public static IGraphQLBuilder Configure<TOptions>(this IGraphQLBuilder graphQLBuilder, Action<TOptions>? action)
+            where TOptions : class, new()
+            => graphQLBuilder.Configure<TOptions>(action == null ? null : (opt, _) => action(opt));
+        #endregion
+
+        #region - AddSchema -
+        /// <summary>
+        /// Registers <typeparamref name="TSchema"/> within the dependency injection framework. <see cref="ISchema"/> is also
+        /// registered if it is not already registered within the dependency injection framework. Singleton and scoped
+        /// lifetimes are supported.
+        /// </summary>
+        /// <remarks>
+        /// Schemas that implement <see cref="IDisposable"/> of a transient lifetime are not supported, as this will cause a
+        /// memory leak if requested from the root service provider.
+        /// </remarks>
+        public static IGraphQLBuilder AddSchema<TSchema>(this IGraphQLBuilder builder, ServiceLifetime serviceLifetime = ServiceLifetime.Singleton)
+            where TSchema : class, ISchema
+        {
+            if (serviceLifetime == ServiceLifetime.Transient && typeof(IDisposable).IsAssignableFrom(typeof(TSchema)))
+            {
+                // This scenario can cause a memory leak if the schema is requested from the root service provider.
+                // If it was requested from a scoped provider, then there is no reason to register it as transient.
+                // See following link:
+                // https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#disposable-transient-services-captured-by-container
+                throw new InvalidOperationException("A schema that implements IDisposable cannot be registered as a transient service.");
```

Comment:
The same suggestion here about error message.

Reply:
#2785